### PR TITLE
skip test docker if we do not use docker as container runtime.

### PR DESCRIPTION
**Pull request by xlgao-zju**
skip test docker if we do not use docker as container runtime.

Signed-off-by: Xianglin Gao <xlgao@zju.edu.cn>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36216)

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -475,9 +475,11 @@ function start_kubelet {
       # Docker won't run a container with a cidfile (container id file)
       # unless that file does not already exist; clean up an existing
       # dockerized kubelet that might be running.
+      test_docker
```
Comment (ixdy) [inline hack/local-up-cluster.sh]: why is this added here?

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -609,7 +611,9 @@ EOF
 fi
 }
 
-test_docker
+if [[ "${CONTAINER_RUNTIME}" == "docker" ]] || [[ -z "${CONTAINER_RUNTIME}" ]]; then
+  test_docker
```
Comment (ixdy) [inline hack/local-up-cluster.sh]: should there be tests for alternate container runtimes too?

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (xlgao-zju) [inline hack/local-up-cluster.sh]: Cause this [line](https://github.com/xlgao-zju/kubernetes/blob/a79a161c3d40133fb69cad995cada77e8c9a30c8/hack/local-up-cluster.sh#L497) will use docker.

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (xlgao-zju) [inline hack/local-up-cluster.sh]: @ixdy Yes, but how to test `remote` container-runtime or we just skip test when we use `remote`?

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (ixdy) [inline hack/local-up-cluster.sh]: it seems redundant with the `test_docker` call on L631, though.

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (ixdy) [inline hack/local-up-cluster.sh]: good question. :) I don't know the answer. 

minor nit: the `[[ -z "${CONTAINER_RUNTIME}" ]]` branch of this probably won't ever apply and can be removed, since we set `CONTAINER_RUNTIME` to `docker` by default if it's empty.

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -130,6 +130,22 @@ function test_docker {
     fi
 }
 
+function test_rkt {
+    if [[ -n "${RKT_PATH}" ]]; then
+      ${RKT_PATH} list 2> /dev/null 1> /dev/null
+      if [ "$?" != "0" ]; then
+        echo "Failed to successfully run 'rkt list', please verify that $RKT_PATH is the path of rkt binary."
```
Comment (ixdy) [inline hack/local-up-cluster.sh]: minor nit: `${RKT_PATH}` (add { }) to be consistent with the rest of this function.

**Review by ixdy — COMMENTED** [review]
looks good, just a few super-minor nits.

**Comment by xlgao-zju** [comment]
@ixdy Thank you for your nits. And I update this pr with your nits. But I have a question : why the indent between different functions is different such as L87 and L125?

**Comment by ixdy** [comment]
> why the indent between different functions is different such as L87 and L125?

because we haven't strictly enforced a style. A lot of our bash scripts are inconsistent like this.

**Comment by xlgao-zju** [comment]
@ixdy Thank you for your answer. Is this pr ok to test :)

**Comment by ixdy** [comment]
LGTM

ok to test

**Comment by xlgao-zju** [comment]
@k8s-bot unit test this

**Comment by xlgao-zju** [comment]
@ixdy The unit test failed :/

**Comment by ixdy** [comment]
@k8s-bot unit test this

**Comment by xlgao-zju** [comment]
@ixdy gci gke e2e test failed, and don't know why it didn't get merged.

**Comment by ixdy** [comment]
@k8s-bot gci gke e2e test this

@xlgao-zju we're currently under code freeze for the 1.5 release. As soon as that thaws, this will be merged.
